src/galileo/handlers/langchain/async_handler.py

```diff
@@ -0,0 +1,472 @@
+        self.span_params: dict[str, Any] = span_params
+        self.run_id: UUID = run_id
+        self.parent_run_id: Optional[UUID] = parent_run_id
+        self.children: list[str] = []
+
+
+class GalileoAsyncCallback(AsyncCallbackHandler):
+    """
+    Async Langchain callback handler for logging traces to the Galileo platform.
+
+    Attributes
+    ----------
+    _galileo_logger : GalileoLogger
+        The Galileo logger instance.
+    _nodes : dict[UUID, Node]
+        A dictionary of nodes, where the key is the run_id and the value is the node.
+    _start_new_trace : bool
+        Whether to start a new trace when a chain starts. Set this to `False` to continue using the current trace.
+    _flush_on_chain_end : bool
+        Whether to flush the trace when a chain ends.
+    """
+
+    def __init__(
+        self,
+        galileo_logger: Optional[GalileoLogger] = None,
+        start_new_trace: bool = True,
+        flush_on_chain_end: bool = True,
+    ):
+        self._galileo_logger: GalileoLogger = galileo_logger or galileo_context.get_logger_instance()
+        self._start_new_trace: bool = start_new_trace
+        self._flush_on_chain_end: bool = flush_on_chain_end
+        self._nodes: dict[str, Node] = {}
+        self._root_node: Optional[Node] = None
+
+    async def _commit(self) -> None:
+        """
+        Commit the nodes to the trace using the Galileo Logger. Optionally flush the trace.
+        """
+        if not self._nodes:
+            _logger.warning("No nodes to commit")
+            return
+
+        root = self._root_node
+        if root is None:
+            _logger.warning("Unable to add nodes to trace: Root node not set")
+            return
+
+        root_node = self._nodes.get(str(root.run_id))
+        if root_node is None:
+            _logger.warning("Unable to add nodes to trace: Root node does not exist")
+            return
+
+        if self._start_new_trace:
+            self._galileo_logger.start_trace(input=serialize_to_str(root_node.span_params.get("input", "")))
+
+        await self._log_node_tree(root_node)
+
+        # Conclude the trace with the root node's output
+        root_output = root_node.span_params.get("output", "")
+
+        if self._start_new_trace:
+            # If we started a new trace, we need to conclude it
+            self._galileo_logger.conclude(output=serialize_to_str(root_output))
+
+        if self._flush_on_chain_end:
+            # Upload the trace to Galileo
+            await self._galileo_logger.async_flush()
+
+        # Clear nodes after successful commit
+        self._nodes.clear()
+        self._root_node = None
+
+    async def _log_node_tree(self, node: Node) -> None:
+        """
+        Log a node and its children recursively.
+
+        Parameters
+        ----------
+        node : Node
+            The node to log.
+        """
+        is_workflow_span = False
+        input = node.span_params.get("input", "")
```

Comment:
```suggestion
        input_ = node.span_params.get("input", "")
```
cause it shadows built-in input function